### ISSUE TITLE
[BUG] Delta table out of order after caused by mergeSchema

### ISSUE THREAD

**Issue by synchris** [delta-io/delta#1428]
## Bug
Delta table out of order after caused by mergeSchema

### Describe the problem
Append rows with the first schema (Delta streaming)
Create aggregate (aggregate is correct)
Append rows using a new schema (Delta streaming)
Create aggregate (aggregate is wrong)

#### Steps to reproduce
Append rows with the first schema (Delta streaming)
Create aggregate (aggregate is correct)
Append rows using a new schema (Delta streaming)
Create aggregate (aggregate is wrong)

#### Observed results

The aggregate with the new schema is inconsistent (wrong)

#### Expected results

The aggregate results should be the same as with the previous schema

### Environment information

* Delta Lake version: 1.0.0
* Spark version: 3.1.3
* Scala version: 2.12

### Willingness to contribute

The Delta Lake Community encourages bug fix contributions. Would you or another member of your organization be willing to contribute a fix for this bug to the Delta Lake code base?

- [ ] Yes. I can contribute a fix for this bug independently.
- [ ] Yes. I would be willing to contribute a fix for this bug with guidance from the Delta Lake community.
- [ ] No. I cannot contribute a bug fix at this time.

**Comment by zsxwing** [comment]
Do you have a concrete example to show the issue? It's hard to get the problem you raised right now.

**Comment by synchris** [comment]
Yes I have a concrete example, I will also edit the description to explain further the inconsistency.

**Comment by foss-contributor** [comment]
@synchris, thank you for your contribution! This issue has been inactive for over 18 months and will be closed in one week. To keep it open, please leave a comment on this issue or contact us in the `#contributors` channel on [Delta Lake Slack](https://go.delta.io/slack). We appreciate your involvement in Delta Lake!
